types/react/index.d.ts

```diff
@@ -26,6 +26,7 @@ declare namespace React {
     // ----------------------------------------------------------------------
 
     type ReactType = string | ComponentClass<any> | StatelessComponent<any>;
+    type ReactComponent<P> = ComponentClass<P> | StatelessComponent<P>;
```

Comment:
maybe `ComponentType<P>` to be consistent with `ReactType`, and then use it above as `ComponentType<{}>`